admin/resources/js/components/H2Components/Button.tsx

```diff
@@ -1,9 +1,15 @@
 import React from "react";
 
 export interface ButtonProps {
+  /** The child elements to appear within the parent element. */
   children: React.ReactNode;
+  /** The style type of the element. */
   color: "primary" | "secondary" | "cta" | "white";
+  /** The style mode of the element. */
   mode: "solid" | "outline" | "inline";
+  /** Determines whether the element should be block level and 100% width. */
+  block?: boolean;
+  /** The click handler for the element. */
   onClick?: (event: React.MouseEvent<HTMLButtonElement>) => void;
 }
```

Comment:
Instead of adding props ad-hoc as required, I'd suggest having ButtonProps extend HTMLButtonElement. Then, in the Button component args, add `...rest`, and add all those to the button element with `<button {...rest}>`.

This would allow us to pass any data- attributes we want in from the parent, and the style prop, and anything else we could normally put on a button element.